### DISCUSSION TITLE
How to create Model Factories?

### DISCUSSION THREAD

**Discussion by 1jason1** [artkonekt/concord#14] [Q&A]
Hello, ive found your concord package and find it very interesting. Been trying to make a proof of concept with it but ive ran into an issue. How can I add HasFactory to my models and then use during testing? The path for the models gets appended to the Default laravel database/factories path.

**Comment by fulopattila122** [comment]
> The path for the models gets appended to the Default laravel database/factories path.

I don't exactly understand. Could you please add more details?

**Reply by 1jason1** [reply]
So i created an external Module package "Integrations", and added to the application via composer. All worked well, but in the unit tests inside of the Application the Models included from the Concord/Module dont link back to their location within the Module. Laravel is looking for them in the database/factories location. 

Example code below does not work, and produces this:  _Error: Class 'Database\Factories\Enrgio\Integrations\Models\IntegrationFactory' not found_

![image](https://user-images.githubusercontent.com/5578451/156603196-2693a98a-2d39-4d04-b0be-ea3c757e962a.png)

This works, but would rather use a factory.
![image](https://user-images.githubusercontent.com/5578451/156603929-a898f9d7-1ee7-490f-ad46-adbadabf7997.png)

**Reply by 1jason1** [reply]
Also, this is the module file structue i used. 
<img width="325" alt="image" src="https://user-images.githubusercontent.com/5578451/156605995-d6ebff52-591f-45ea-9e06-f10cde58a728.png">

**Comment by fulopattila122** [comment]
As far as I see, this is a plain namespacing issue: IntegrationFactory needs to be placed right next to the Model, or if you stick to the current location, set up the right namespace in the module's composer.json (autoload, PSR-4) and use it with the corresponding namespace in the model.

**Comment by 1jason1** [comment]
Thanks for the response, do you have an example of how I would set the right namespace in the modules composer.json? Apprciate you help.

**Reply by fulopattila122** [reply]
Just put the factory Class in the same folder as the model and make sure it has the same namespace.

This is PHP coding, has nothing to do with Concord, I suggest googling for PSR-4 as a start 🤘

**Reply by fulopattila122** [reply]
Also: https://getcomposer.org/doc/04-schema.md#autoload

**Reply by 1jason1** [reply]
Thank you!

**Reply by fulopattila122** [reply]
> Thank you!

Let me know how did it go :+1:
